pkg/cmd/unjoin/exec.go

```diff
@@ -0,0 +1,131 @@
+// Copyright Contributors to the Open Cluster Management project
+package unjoin
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+
+	klusterletclient "open-cluster-management.io/api/client/operator/clientset/versioned"
+	appliedwrokclient "open-cluster-management.io/api/client/work/clientset/versioned"
+	"open-cluster-management.io/clusteradm/pkg/helpers"
+	"open-cluster-management.io/clusteradm/pkg/helpers/apply"
+)
+
+func (o *Options) complete(cmd *cobra.Command, args []string) (err error) {
+	klog.V(1).InfoS("unjoin  options:", "dry-run", o.ClusteradmFlags.DryRun, "cluster", o.clusterName, o.outputFile)
+
+	o.values = Values{
+		ClusterName: o.clusterName,
+	}
+	klog.V(3).InfoS("values:", "clusterName", o.values.ClusterName)
+	return nil
+}
+
+func (o *Options) validate() error {
+	if o.values.ClusterName == "" {
+		return fmt.Errorf("name is missing")
+	}
+	return nil
+}
+
+func (o *Options) run() error {
+
+	// Delete the applied resource in the Managed cluster
+	nameSpace := "default"
+	output := make([]string, 0)
+	fmt.Printf("Remove applied resources in the managed cluster %s ... \n", o.clusterName)
+
+	//Delete Klusterlet CR resources firstly
+	f := o.ClusteradmFlags.KubectlFactory
+	config, err := f.ToRESTConfig()
+	if err != nil {
+		log.Fatal(err)
+	}
+	kubeClient, apiExtensionsClient, _, err := helpers.GetClients(f)
+	if err != nil {
+		log.Fatal(err)
+	}
+	appliedWorkClient, err := appliedwrokclient.NewForConfig(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if IsAppliedManifestWorkExist(appliedWorkClient) {
+
+		log.Fatal("AppliedManifestWork exist on the managed cluster, uninstalling the klusterlet will cause that the manifestworks on hub cannot be cleaned.")
+	} else {
+		//Create clusterlet client
+		klusterletClient, err := klusterletclient.NewForConfig(config)
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = klusterletClient.OperatorV1().Klusterlets().Delete(context.Background(), "klusterlet", metav1.DeleteOptions{})
+		if err != nil {
+			log.Fatal(err)
+		}
+		if !o.ClusteradmFlags.DryRun {
+			b := retry.DefaultBackoff
+			b.Duration = 1 * time.Minute
+
+			err = WaitResourceToBeDelete(context.Background(), klusterletClient, "klusterlet", b)
+			if err != nil {
+				log.Println("Recource Klusterlet have been deleted firstly.")
+
+			}
+		}
+	}
+
+	//Delete the other applied resources
+	if o.purgeOperator {
+		kubeClient.AppsV1().Deployments(nameSpace).Delete(context.Background(), "klusterlet", metav1.DeleteOptions{})
+	}
+	apiExtensionsClient.ApiextensionsV1().CustomResourceDefinitions().Delete(context.Background(), "klusterlets.operator.open-cluster-management.io", metav1.DeleteOptions{})
```

Comment:
these resources are all relates to operator, so put this in if o.purgeOperator

Reply:
Refreshed again